query-engine/driver-adapters/js/adapter-libsql/src/libsql.ts

```diff
@@ -112,11 +128,19 @@ export class PrismaLibSQL extends LibSqlQueryable<StdClient> implements DriverAd
     const tag = '[js::startTransaction]'
     debug(`${tag} options: %O`, options)
 
-    const tx = await this.client.transaction('deferred')
-    return ok(new LibSqlTransaction(tx, options))
+    const release = await this[LOCK_TAG].acquire()
+
+    try {
+      const tx = await this.client.transaction('deferred')
+      return ok(new LibSqlTransaction(tx, options, release))
+    } catch (e) {
+      release()
+      throw e
+    }
```

Comment:
not to self: maybe add a comment explaining why `finally` would be incorrect here just in case?